### PR TITLE
Airneis 43:  Database migration

### PR DESCRIPTION
**Pull request by cedricgautier**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,3 +21,7 @@ services:
       - .env.local
     ports:
       - "5432:5432"
+    environment:
```
Comment (barnabepilliaudin) [inline docker-compose.yml]: Should the db be moved to the dev docker-compose. As we will be using neon for production ?

**Review by BarnabePILLIAUDIN — APPROVED** [review]
LGTM!

File: back/config/db.go [hunk 2]
```diff
@@ -1,29 +1,94 @@
...
+
+	default:
+		database, err = InitNeonDatabase(config.Handler.VaultClient)
+		if err != nil {
+			return errors.Wrapf(err, "InitDatabase, unable to load database")
+		}
+
+	}
+
+	log.Info("Database has been loadedddd")
```
Comment (Magali-D) [inline back/config/db.go]:
```suggestion
	log.Info("Database has been loaded")
```

**Review by Magali-D — APPROVED** [review]

**Comment by Magali-D** [comment]
Working on my side

File: back/model/Product.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	CategoryID              uint           `json:"categoryID"`
+	Category                Category       `json:"category"`
+	PriceWithoutTaxes       int            `json:"PriceWithoutTaxes"`
+	PriceWithTaxes          int            `json:"priceWithTaxes"`
+	Description             string         `json:"description"`
+	ThumbnailUrl            string         `json:"thumbnailUrl"`
+	OutOfStock              string         `json:"outOfStock" gorm:"default:false"`
+	Materials               []Material     `json:"materials" gorm:"many2many:materials_of_products"`
+	IsHighlanderOfTheMoment bool           `json:"isHighlanderOfTheMoment" gorm:"default:false"`
+	ProductImages           []ProductImage `json:"productImages" gorm:"many2many:images_of_products"`
```
Comment (KyDev25) [inline back/model/Product.go:19]:
```suggestion
        OrderOfPriority           int                     `json:"orderOfPriority"`
	ProductImages           []ProductImage `json:"productImages" gorm:"many2many:images_of_products"`
```

**Review by KyDev25 — CHANGES_REQUESTED** [review]
We forgot to add a field to determine the display priority for products in a category.

**Review by KyDev25 — APPROVED** [review]

File: back/config/config.go [hunk 4]
```diff
@@ -91,21 +86,43 @@ func InitializeConfig() (config Config, err error) {
...
-			return Config{}, errors.Wrapf(err, "InitializeConfig, unable to load config")
+			return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to load config")
+		}
+
+		if err = InitDatabase(config, productionEnv); err != nil {
+			return Config{}, errors.Wrapf(err, "buildEnvironmentConfig, unable to initialize production database")
 		}
-		logrus.Info("Production Environment has been loaded")
+
+		log.Info("Production Environment has been loaded")
```
Comment (avetisk) [inline back/config/config.go:122]: Where's my early return? 🤔

**Pull request merged**